### PR TITLE
Cloud gaming

### PR DESCRIPTION
**Pull request by wenbinc-Bin**

### CODE REVIEW

**Comment by wenbinc-Bin** [comment]
@xhaihao @feiwan1 @tong1wu Could you please help to review this patchset?
Thanks

**Comment by wenbinc-Bin** [comment]
I remove the bitrate and framerate reset from this patchset.

**Comment by wenbinc-Bin** [comment]
I remove the framerate and bitrate reset from this patchset, and will submit them in other patchset.

File: doc/encoders.texi [hunk 1]
```diff
@@ -3344,6 +3344,9 @@ Following options can be used durning qsv encoding.
 @item @var{b_quant_offset}
 Supported in h264_qsv and hevc_qsv.
 Change these value to reset qsv codec's qp configuration.
+
+@item @var{max_frame_size}
+Change this value to reset qsv codec's MaxFrameSize configuration.
```
Comment (dvrogozh) [inline doc/encoders.texi:3350]: Missing applicability:
```
Supported in h264_qsv and hevc_qsv.
```

File: doc/encoders.texi [hunk 2]
```diff
@@ -3350,6 +3350,12 @@ Change this value to reset qsv codec's MaxFrameSize configuration.
 
 @item @var{gop_size}
 Change this value to reset qsv codec's gop configuration.
+
+@item @var{int_ref_type}
+@item @var{int_ref_cycle_size}
+@item @var{int_ref_qp_delta}
+@item @var{int_ref_cycle_dist}
+Change these value to reset qsv codec's Intra Refresh configuration.
```
Comment (dvrogozh) [inline doc/encoders.texi:3360]: applicability missing:
```
Supported in h264_qsv and hevc_qsv.
```

File: doc/encoders.texi [hunk 3]
```diff
@@ -3356,6 +3356,16 @@ Change this value to reset qsv codec's gop configuration.
...
+
+@item @var{qmax}
+@item @var{qmin}
+@item @var{max_qp_i}
+@item @var{min_qp_i}
+@item @var{max_qp_p}
+@item @var{min_qp_p}
+@item @var{max_qp_b}
+@item @var{min_qp_b}
+Change these value to reset qsv codec's max/min qp configuration.
```
Comment (dvrogozh) [inline doc/encoders.texi:3371]: applicability missing:
```
Supported in h264_qsv and hevc_qsv.
```

**Review by dvrogozh — CHANGES_REQUESTED** [review]
How did you test these? I thought you intended to provide ffmpeg examples update for that?

**Comment by wenbinc-Bin** [comment]
> How did you test these? I thought you intended to provide ffmpeg examples update for that?

I will submit PR for ffmpeg example later. I plan to add ffmpeg example after I add enough dynamic resetting support.

File: doc/encoders.texi (hunk 1 above)
Comment (wenbinc-Bin) [reply doc/encoders.texi:3350]: Done, thanks

File: doc/encoders.texi (hunk 3 above)
Comment (wenbinc-Bin) [reply doc/encoders.texi:3371]: Done, thanks

File: doc/encoders.texi (hunk 2 above)
Comment (wenbinc-Bin) [reply doc/encoders.texi:3360]: Done, thanks

File: doc/encoders.texi (hunk 3 above)
Comment (wenbinc-Bin) [reply doc/encoders.texi:3371]: I verified it only supported on h264, so only add it on h264.

Comment (wenbinc-Bin) [reply doc/encoders.texi:3371]: Also confirmed with VPL team.

File: doc/encoders.texi [hunk 4]
```diff
@@ -3344,6 +3344,35 @@ Following options can be used durning qsv encoding.
...
+@item @var{max_qp_i}
+@item @var{min_qp_i}
+@item @var{max_qp_p}
+@item @var{min_qp_p}
+@item @var{max_qp_b}
+@item @var{min_qp_b}
+Supported in h264_qsv.
+Change these value to reset qsv codec's max/min qp configuration.
+
+@item @var{low_delay_brc}
```
Comment (dvrogozh) [inline doc/encoders.texi:3373]: This can be runtime setting? Interesting...

**Review by dvrogozh — APPROVED** [review]

**Comment by wenbinc-Bin** [comment]
submit to upstream:[https://patchwork.ffmpeg.org/project/ffmpeg/list/?series=7246]( https://patchwork.ffmpeg.org/project/ffmpeg/list/?series=7246 )

**Pull request closed without merging**